### PR TITLE
Add Inputs/Outputs check interface

### LINKED ISSUE #4434: Add HasInputs/HasOutputs in shape interface
Maybe we need to add interfaces `HasInputs/HasOutputs` in `InferShapeContextBase`

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #4434

### CODE REVIEW

File: paddle/framework/shape_inference.h [hunk 1]
```diff
@@ -24,6 +24,14 @@ class InferShapeContextBase {
   virtual ~InferShapeContextBase() {}
   virtual bool HasInput(const std::string &name) const = 0;
   virtual bool HasOutput(const std::string &name) const = 0;
+  bool HasInputs(const std::string &name) const {
+    const std::vector<std::string> &names = Inputs(name);
+    return names.size() > 0;
+  }
+  bool HasOutputs(const std::string &name) const {
```
Comment (typhoonzero) [inline paddle/framework/shape_inference.h]: This can be `inline`

File: paddle/framework/shape_inference.h [hunk 2]
```diff
@@ -24,6 +24,14 @@ class InferShapeContextBase {
   virtual ~InferShapeContextBase() {}
   virtual bool HasInput(const std::string &name) const = 0;
   virtual bool HasOutput(const std::string &name) const = 0;
+  bool HasInputs(const std::string &name) const {
+    const std::vector<std::string> &names = Inputs(name);
```
Comment (jacquesqiao) [inline paddle/framework/shape_inference.h]: I think we need to check if these variables are all in the scope~

Comment (Yancey1989) [reply paddle/framework/shape_inference.h]: Done.

File: paddle/framework/shape_inference.h (hunk 1 above)
Comment (Yancey1989) [reply paddle/framework/shape_inference.h]: Check var in scope, so we need a virtual function.

**Pull request merged**